rust/lance-io/src/object_reader.rs

```diff
@@ -166,3 +169,138 @@ impl Reader for CloudObjectReader {
         .await
     }
 }
+
+/// A reader for a file so small, we just eagerly read it all into memory.
+///
+/// When created, it represents a future that will read the whole file into memory.
+///
+/// On the first read call, it will start the read. Multiple threads can call read at the same time.
+///
+/// Once the read is complete, any thread can call read again to get the result.
+#[derive(Debug)]
+pub struct SmallReader {
+    path: Path,
+    block_size: usize,
```

Comment:
Why is this part of the struct and not a constant?

Reply:
I don't know why I didn't think of that...